src/models/converters/transactions/safe_app_info.rs

```diff
@@ -0,0 +1,41 @@
+use crate::models::service::transactions::summary::SafeAppInfo;
+use crate::providers::info::InfoProvider;
+use serde::Deserialize;
+
+pub fn safe_app_info_from(
+    origin: &str,
+    info_provider: &mut dyn InfoProvider,
+) -> Option<SafeAppInfo> {
+    let origin_internal = serde_json::from_str::<OriginInternal>(origin).ok();
+    origin_internal
+        .as_ref()
+        .and_then(|origin| origin.to_safe_app_info(info_provider))
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub(super) struct OriginInternal {
+    pub(super) url: String,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub(super) struct Manifest {
+    pub(super) name: String,
+    pub(super) description: String,
+    #[serde(rename(deserialize = "iconPath"))]
+    pub(super) icon_path: String,
+}
+
+impl OriginInternal {
+    fn to_safe_app_info(&self, info_provider: &mut dyn InfoProvider) -> Option<SafeAppInfo> {
```

Comment:
Wouldn't it be nicer to return a Result here that we then convert to an Option in the `safe_app_info_from` method? (would allow it to use the `?` to unwrap)